src/EditorFeatures/Test/CodeGeneration/AbstractCodeGenerationTests.cs

```diff
@@ -14,6 +17,79 @@ namespace Microsoft.CodeAnalysis.Editor.UnitTests.CodeGeneration
     [UseExportProvider]
     public abstract class AbstractCodeGenerationTests
     {
+        private static SyntaxNode Simplify(
+            AdhocWorkspace workspace,
+            SyntaxNode syntaxNode,
+            string languageName)
+        {
+            var projectId = ProjectId.CreateNewId();
+
+            var project = workspace.CurrentSolution
+                .AddProject(projectId, languageName, $"{languageName}.dll", languageName).GetProject(projectId);
+
+            var normalizedSyntax = syntaxNode.NormalizeWhitespace().ToFullString();
+            var document = project.AddMetadataReference(TestReferences.NetFx.v4_0_30319.mscorlib)
+                .AddDocument("Fake Document", SourceText.From(normalizedSyntax));
+
+            var annotatedDocument = document.WithSyntaxRoot(
+                    document.GetSyntaxRootAsync().Result.WithAdditionalAnnotations(Simplification.Simplifier.Annotation));
+
+            var annotatedRootNode = annotatedDocument.GetSyntaxRootAsync().Result;
+
+            var simplifiedDocument = Simplification.Simplifier.ReduceAsync(annotatedDocument).Result;
+
+            var rootNode = simplifiedDocument.GetSyntaxRootAsync().Result;
+
+            return rootNode;
+        }
+
+        private static SyntaxNode WrapExpressionInBoilerplate(Func<SyntaxGenerator, SyntaxNode> nodeCreator, SyntaxGenerator codeDefFactory)
```

Comment:
Feels like the nodeCreator is superfluous.  Just pass the SyntaxGenerator and the generated expression to put in the local-decl-statement.